adadamp/_dist.py

```diff
@@ -57,13 +78,23 @@ def gradient(
     where `l` is the loss function for a single example.
 
     """
-    if idx is not None:
-        inputs = inputs[idx]
-        targets = targets[idx]
-    inputs, targets = inputs.to(device), targets.to(device)
+    # Dirty hack! This function should not be in this file.
+    #
+    # However, client.scatter(train_set) created problems with
+    # client.scatter(model)
+    # See https://github.com/dask/distributed/issues/3807 for more detail
```

Comment:
I'd move this to `_get_fashionmnist`'s docstring, and point to it from `dist_example.gradients` and the lines above.